lib/babe/verify_test.go

```diff
@@ -1,4 +1,1296 @@
+		},
+		{
+			name:     "invalid preruntime digest data",
+			verifier: verifier{},
+			header:   header2,
+			expErr:   errors.New("failed to verify pre-runtime digest: EOF, field: 0"),
+		},
+		{
+			name:     "invalid seal length",
+			verifier: *babeVerifier,
+			header:   header3,
+			expErr:   errors.New("invalid signature length"),
+		},
+		{
+			name:     "invalid seal signature - primary",
+			verifier: *babeVerifier2,
+			header:   header4,
+			expErr:   ErrBadSignature,
+		},
+		{
+			name:     "invalid seal signature - secondary plain",
+			verifier: *babeVerifier3,
+			header:   header5,
+			expErr:   ErrBadSignature,
+		},
+		{
+			name:     "invalid seal signature - secondary vrf",
+			verifier: *babeVerifier4,
+			header:   header6,
+			expErr:   ErrBadSignature,
+		},
+		{
+			name:     "valid digest items, getAuthorityIndex error",
+			verifier: *babeVerifier5,
+			header:   header7,
+		},
+		{
+			name:     "get header err",
+			verifier: *babeVerifier6,
+			header:   header7,
+		},
+		{
+			name:     "equivocate - primary",
+			verifier: *babeVerifier7,
+			header:   header7,
+			expErr:   ErrProducerEquivocated,
+		},
+		{
+			name:     "equivocate - secondary plain",
+			verifier: *babeVerifier8,
+			header:   header8,
+			expErr:   ErrProducerEquivocated,
+		},
+		{
+			name:     "equivocate - secondary vrf",
+			verifier: *babeVerifier9,
+			header:   header9,
+			expErr:   ErrProducerEquivocated,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &tt.verifier
+			err := b.verifyAuthorshipRight(tt.header)
+			if tt.expErr != nil {
+				assert.EqualError(t, err, tt.expErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+		})
+	}
+}
+
+func TestVerificationManager_getConfigData(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockBlockState := NewMockBlockState(ctrl)
+	mockEpochStateEmpty := NewMockEpochState(ctrl)
+	mockEpochStateHasErr := NewMockEpochState(ctrl)
+	mockEpochStateGetErr := NewMockEpochState(ctrl)
+
+	mockEpochStateEmpty.EXPECT().HasConfigData(gomock.Eq(uint64(0))).Return(false, nil)
+	mockEpochStateHasErr.EXPECT().HasConfigData(gomock.Eq(uint64(0))).Return(false, errors.New("no ConfigData"))
+	mockEpochStateGetErr.EXPECT().HasConfigData(gomock.Eq(uint64(0))).Return(true, nil)
+	mockEpochStateGetErr.EXPECT().GetConfigData(gomock.Eq(uint64(0))).Return(nil, errors.New("cant get ConfigData"))
+
+	vm0, err := NewVerificationManager(mockBlockState, mockEpochStateEmpty)
+	assert.NoError(t, err)
+	vm1, err := NewVerificationManager(mockBlockState, mockEpochStateHasErr)
+	assert.NoError(t, err)
+	vm2, err := NewVerificationManager(mockBlockState, mockEpochStateGetErr)
+	assert.NoError(t, err)
+	tests := []struct {
+		name   string
+		vm     *VerificationManager
+		epoch  uint64
+		exp    *types.ConfigData
+		expErr error
+	}{
+		{
+			name:   "cant find ConfigData",
+			vm:     vm0,
+			epoch:  0,
+			expErr: errors.New("cannot find ConfigData for epoch"),
```

Comment:
can you use named errors here? eg. 
`var errNoConfigData = errors.New("cannot find ConfigData for epoch")` and assert that the error is of type `errNoConfigData`, same for the others here